staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -115,11 +116,15 @@ func (f *FieldManager) Update(liveObj, newObj runtime.Object, manager string) (r
 	internal.RemoveObjectManagedFields(newObjVersioned)
 	newObjTyped, err := f.typeConverter.ObjectToTyped(newObjVersioned)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create typed new object: %v", err)
+		// Return newObj and just by-pass fields update. This really shouldn't happen.
+		klog.Errorf("[SHOULD NOT HAPPEN] failed to create typed new object: %v", err)
+		return newObj, nil
```

Comment:
This seems mostly safe to me, but if newObjVersioned is just a pointer to newObj (in the case where `f.toVersioned(newObj)` is a no-op) then we would be wiping the managed fields. It wouldn't really be an issue though because this should never happen if managed fields is working correctly

Reply:
Should we make a copy of the object when it's a no-op?

Reply:
Looks like it already does a copy, that seems to be the difference between `ConvertToVersion` and `UnsafeConvertToVersion`